core/lib/resources/presence/presence_manager.js

```diff
@@ -39,13 +40,30 @@ PresenceManager.prototype.setup = function() {
   });
 
   // save so you removeListener on destroy
-  this.sentryListener = function(sentry) {
+  this.sentryListener = function (sentry) {
+    if (manager.destroying) return;
+
     var clientIds = store.clientsForSentry(sentry);
 
     var chain = function (clientIds) {
+      var immediateObject;
       var clientId = clientIds.pop();
 
-      if (!clientId) return;
+      // Pick up any new users/clients that have subscribed to this resource
+      if (!clientId && !manager.destroying) {
+        clientIds = store.clientsForSentry(sentry);
```

Comment:
A problem with only one call to clientsForSentry() is that it creates a cached set of data that ages while the chained calls to the sentryListener are firing.  When we pop the last entry from that initial cached set, we should do one more check to see whether additional users/clients have subscribed to the resource - otherwise, they will not be notified.